File: validate/cc_corpus/fixtures/equivalence.ts
function no_branches(x: number): number {
    return x + 1;
}

function single_if(x: number): number {
    if (x > 0) return 1;
    return 0;
}

function if_else_if(x: number): number {
    if (x > 0) return 1;
    else if (x < 0) return -1;
    else return 0;
}

function nested_if(x: number, y: number): number {
    if (x > 0) {
        if (y > 0) return 1;
    }
    return 0;
}

function for_loop(lst: number[]): void {
    for (let i = 0; i < lst.length; i++) {}
    return;
}

function for_with_if(lst: number[]): number {
    for (let i = 0; i < lst.length; i++) {
        if (lst[i] > 0) return lst[i];
    }
    return 0;
}

function while_loop(n: number): number {
    while (n > 0) { n--; }
    return n;
}

function try_catch(x: number): number {
    try {
        return Math.floor(1 / x);
    } catch(e) {
        if (x === 0) return -1;
        return 1;
    }
    return 0;
}

function boolean_ops(a: boolean, b: boolean, c: boolean): number {
    if (a && b || c) return 1;
    return 0;
}

function early_return(val: number): number {
    if (val < 0) return -1;
    if (val === 0) return 0;
    return 1;
}

function ternary(x: boolean, a: number, b: number): number {
    return x ? a : b;
}

function switch_case(x: number): string {
    switch(x) {
        case 1: return 'a';
        case 2: return 'b';
        default: return 'z';
    }
}

File: validate/cc_corpus/fixtures/equivalence.py
def no_branches(x):
    return x + 1


def single_if(x):
    if x > 0:
        return 1
    return 0


def if_else_if(x):
    if x > 0:
        return 1
    elif x < 0:
        return -1
    else:
        return 0


def nested_if(x, y):
    if x > 0:
        if y > 0:
            return 1
    return 0


def for_loop(lst):
    for i in lst:
        pass
    return


def for_with_if(lst):
    for i in lst:
        if i > 0:
            return i
    return 0


def while_loop(n):
    while n > 0:
        n -= 1
    return n


def try_catch(x):
    try:
        return 1 // x
    except ZeroDivisionError:
        if x == 0:
            return -1
    return 0


def boolean_ops(a, b, c):
    if a and b or c:
        return 1
    return 0


def early_return(val):
    if val < 0:
        return -1
    if val == 0:
        return 0
    return 1


def ternary(x, a, b):
    if x:
        return a
    else:
        return b


def switch_case(x):
    if x == 1:
        return 'a'
    elif x == 2:
        return 'b'
    elif x == 3:
        return 'c'
    else:
        return 'z'
